src/components/hint/index.js

```diff
@@ -398,7 +398,7 @@ class Hint extends BaseComponent {
   }
 
   render() {
-    const {visible, onBackgroundPress} = this.props;
+    const {visible, onBackgroundPress, testID} = this.props;
```

Comment:
couple of things regarding testID (: 

testID is an identifier we use for e2e tests and it has to be unique throughout the render tree. 
That means that if two elements will have the same testID (in a screen) the test will fail. 

If you'll search in the code of this component, you'll see we are already passing the testID. 
In these cases, when we need to tag more than one element in the component with a testID. we just concatenate the user's `testID` with the element type/name/id. 
For instance in your case it can be 
`${testID}.modal` or `${testID}.background`, etc... 

Another thing that is worth noticing in this case is that you're passing the testID to the `Modal` component and I'm not sure the Modal handles the `testID`. 
see this: https://stackoverflow.com/questions/50117265/detox-testing-visibility-of-modal-in-react-native
So we might want to treat the Modal component as well.